### PR TITLE
New translation to spanish language.

### PR DESCRIPTION
**Pull request by huguei**
Here are the new two files es.json and gui-faq-es.md. Please I ask you for the rest of steps (typescripts and other json files).

### CODE REVIEW

**Comment by blacksponge** [comment]
Hi, thanks for the translation!
The steps are those described in the [translation guide](https://github.com/zonemaster/zonemaster-gui/blob/develop/docs/TranslationGuide.md), in you case they should be the following:
* Update package.json and add `es` to the language list
https://github.com/zonemaster/zonemaster-gui/blob/d5ea51931d200f0b78da9f1c37948819fe313684/package.json#L14
* Add `import 'moment/locale/es';` after the other `import 'moment/locale/*'` in app.module.ts
https://github.com/zonemaster/zonemaster-gui/blob/d5ea51931d200f0b78da9f1c37948819fe313684/src/app/app.module.ts#L35-L40
* Add the language name (`'es': 'Español',` if I am not making any mistakes) in common.ts 
https://github.com/zonemaster/zonemaster-gui/blob/d5ea51931d200f0b78da9f1c37948819fe313684/src/environments/common.ts#L8-L15
* Also in common.ts, add the language code to the enabled language list
https://github.com/zonemaster/zonemaster-gui/blob/d5ea51931d200f0b78da9f1c37948819fe313684/src/environments/common.ts#L16
* Update the example configuration file to add the language code to the enabled language list
https://github.com/zonemaster/zonemaster-gui/blob/d5ea51931d200f0b78da9f1c37948819fe313684/src/assets/app.config.sample.json#L5
* Create the file `e2e/FR05-es.e2e-spec.ts` with the following content
```ts
import { by, browser, element, ExpectedConditions } from 'protractor';

import { Utils } from './utils/app.utils';

describe('Zonemaster test FR05-es - [Supports Spanish language]', () => {

  const utils = new Utils();

  beforeAll(async () => {
    await utils.goToHome();
  });

  it('should have French language button', () => {
    expect(element(by.xpath('//a[@lang="es"]')).isPresent()).toBe(true);
  });

  it('should switch to French', async () => {
    await utils.setLang('fr');
    await browser.wait(() => ExpectedConditions.presenceOf(element(by.xpath('//h1[.="Nombre de dominio"]'))), 5 * 1000);
    const selectedLang = element.all(by.css('nav div.lang a.selected'));
    expect(selectedLang.count()).toBe(1);
    expect(selectedLang.first().getAttribute('lang')).toBe('es');
  });
});
```
* Update the default languages in the documentation file Configuration.md
https://github.com/zonemaster/zonemaster-gui/blob/d5ea51931d200f0b78da9f1c37948819fe313684/docs/Configuration.md#L25-L26
* Add Spanish to the supported languages in TranslationGuide.md
https://github.com/zonemaster/zonemaster-gui/blob/d5ea51931d200f0b78da9f1c37948819fe313684/docs/TranslationGuide.md#L20-L25

Feel free to ask for clarification if I wasn't clear somewhere.

**Comment by huguei** [comment]
Thanks @blacksponge . In private talks with @matsduf he offered to take care of this steps. I see no problem to do it myself, but I'll wait him to not step into each other.

**Comment by huguei** [comment]
Done. I've just added to the PR the typescript, test and docs modifications.

File: src/environments/common.ts [hunk 1]
```diff
@@ -8,11 +8,12 @@ export const common = {
...
     'da': 'Dansk',
     'en': 'English',
+    'es': 'Español',
     'fi': 'Suomi',
     'fr': 'Français',
     'nb': 'Norsk (bokmål)',
     'sv': 'Svenska'
   },
-  enabledLanguages: [ 'da', 'en', 'fi', 'fr', 'nb', 'sv' ],
+  enabledLanguages: [ 'da', 'en', 'fi', 'fr', 'nb', 'sv', 'es' ],
```
Comment (matsduf) [inline src/environments/common.ts:17]: Please put "es" between "en" and "fi" (alphabetical order)

**Review by matsduf — CHANGES_REQUESTED** [review]
Besides the non-alphabetical order in common.ts I found yet another file that needs addition of "es", i.e. `angular.json`
```
            "allowedCommonJsDependencies": [
              "moment/locale/da",
              "moment/locale/fi",
              "moment/locale/fr",
              "moment/locale/nb",
              "moment/locale/sv"
            ],
```
The file is not documented in the translation guide so I will update there.

Besides that it looks fine. Good and quick job!

**Comment by huguei** [comment]
Last requested changes were done.

**Comment by matsduf** [comment]
@huguei, thanks. Due to another pull request being merged to the develop branch there is a technical conflict. Either you can resolve it or I can do that.

File: src/assets/app.config.sample.json [hunk 2]
```diff
@@ -4,4 +4,5 @@
     "defaultLanguage": "en",
     "enabledLanguages": [ "da", "en", "es", "fi", "fr", "nb", "sv" ],
     "logo": "assets/images/zonemaster_logo_2021_color.png"
```
Comment (blacksponge) [inline src/assets/app.config.sample.json:6]: There is a missing comma at the end of this line

**Comment by matsduf** [comment]
@blacksponge, the Travis error does not look like it is PR related. Is it?

**Comment by blacksponge** [comment]
No, it is caused by some rate limiting on Github API when updating the webdriver. I have run the test locally and they pass successfully.

> [12:32:40] I/config_source - curl -o/home/travis/build/zonemaster/zonemaster-gui/node_modules/webdriver-manager/selenium/gecko-response.json https://api.github.com/repos/mozilla/geckodriver/releases
> (node:6449) UnhandledPromiseRejectionWarning: Error: Failed to make Github request, rate limit reached.

**Comment by matsduf** [comment]
> No, it is caused by some rate limiting on Github API when updating the webdriver.

Github? Not Travis?

**Comment by blacksponge** [comment]
The github API is called from travis when updating the webdriver at this step of the CI script
https://github.com/zonemaster/zonemaster-gui/blob/10a273f6d8a5ed02b6ea31cd335c555b672ef885/.travis.yml#L14

**Comment by blacksponge** [comment]
I have restarted the tests, it works now.

**Comment by matsduf** [comment]
We defer merging until the Spanish translation has been merged in Engine and configuration has been merged in Backend.

**Comment by mattias-p** [comment]
@matsduf Are there any more changes that need to be made to this PR before it can be merged? You're still marked as "Request changes".

**Comment by matsduf** [comment]
> @matsduf Are there any more changes that need to be made to this PR before it can be merged? You're still marked as "Request changes".

No, but we should start with Engine.

**Review by matsduf — APPROVED** [review]

**Pull request merged**